Assets/Scripts/GrappleGunBehavior.cs

```diff
@@ -92,48 +69,71 @@ private void LateUpdate()
+        Vector3[] renderPositions = {gunTip.position, currentTarget.GetGrapplePoint()};
         lineRenderer.SetPositions(renderPositions);
     }
 
     private void FixedUpdate()
     {
         ReticleEffect();
-        if (!grappling) return;
+        if (currentTarget == null) return;
         if (GrappleBroken()) StopGrapple(); // TODO: add particle effects, SFX, and/or animation to indicate what happened
         else
         {
             var tugForce = ComputeTugForce();
-            playerRb.AddForce(tugForce, ForceMode.Force);
+            LevelManager.PlayerRb.AddForce(tugForce, ForceMode.Force);
         }
     }
 
     private bool GrappleBroken()
     {
+        if (currentTarget == null) throw new InvalidOperationException("Not currently grappling.");
         var grappleStart = lineRenderer.GetPosition(0);
         var grappleEnd = lineRenderer.GetPosition(1);
         var grappleDirection = grappleEnd - grappleStart;
         if (Physics.Raycast(grappleStart, grappleDirection, out var hit, grappleDirection.magnitude, grappleableStuff))
         {
-            if (grappledRetractable && hit.collider.transform == grappledObj) return false;
-            return Vector3.Distance(hit.point, GrapplePoint()) > 0.1f;
+            if (currentTarget.IsRetractable && hit.collider.transform == currentTarget.Transform) return false;
+            return Vector3.Distance(hit.point, currentTarget.GetGrapplePoint()) > 0.1f;
         }
         return false;
     }
 
     private Vector3 ComputeTugForce()
     {
-        var grappleDirection = GrapplePoint() - player.transform.position;
+        if (currentTarget == null) throw new InvalidOperationException("Not currently grappling.");
+        var grappleDirection = currentTarget.GetGrapplePoint() - LevelManager.Player.position;
         var tugStrength = grappleDirection.magnitude / maxGrappleLength * maxRetractionForce;
         var tugForce = grappleDirection.normalized * tugStrength;
         return tugForce;
     }
 
-    // TODO: make this private if nothing outside the class uses it
-    public Vector3 GrapplePoint()
+    private class GrappleTarget
     {
-        if (!grappledObjOffset.HasValue) throw new InvalidOperationException("There is no active grapple point.");
-        return grappledObj.position + grappledObjOffset.Value;
+        public Transform Transform { get; }
+        public bool IsRetractable { get; }
+        
+        private Vector3 Offset { get; }
+        private IGrappleResponse[] Responses { get; }
+
+        public GrappleTarget(Transform target, Vector3 grapplePoint)
+        {
+            Transform = target;
+            Offset = grapplePoint - Transform.position;
+            Responses = Transform.GetComponentsInChildren<IGrappleResponse>();
+            foreach (var response in Responses) response.OnGrappleStart();
+            IsRetractable = Responses.Any(res => res is Retractable);
+        }
+
+        public Vector3 GetGrapplePoint()
```

Comment:
Make this a C# property

Reply:
When I try doing `public readonly Vector3 GrapplePoint = Transform.position + Offset;`, it gives me errors saying "Cannot access non-static property 'Transform' in static context", and likewise for `Offset`. Am I using the wrong syntax?